### PR TITLE
order of profile libraries

### PR DESCRIPTION
**Pull request by ahsu1230**
order of profile libraries 
- first Main/Secret on top (only if you view your own profile), then user_created libraries.
- sorted by `member_count` (desc), `last_kept` (desc), `id` (desc)

@andrewconner @2is10

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/model/LibraryRepo.scala [hunk 1]
```diff
@@ -171,13 +170,13 @@ class LibraryRepoImpl @Inject() (
...
-      lm <- libraryMembershipRepo.get.rows if lm.libraryId === lib.id && lm.userId === ownerId && lm.listed
-    } yield lib).sortBy(_.id.desc).drop(page.itemsToDrop).take(page.size)
+      lm <- libraryMembershipRepo.get.rows if lm.libraryId === lib.id && lm.userId === ownerId && lm.listed && lm.state === LibraryMembershipStates.ACTIVE
+    } yield lib).sortBy(x => (x.memberCount.desc, x.lastKept.desc, x.id.desc)).drop(page.itemsToDrop).take(page.size)
     q.list
   }
 
   def getLibrariesOfSelf(userId: Id[User], page: Paginator)(implicit session: RSession): Seq[Library] = {
-    (for (r <- rows if r.ownerId === userId && r.state === LibraryStates.ACTIVE) yield r).sortBy(_.id.desc).drop(page.itemsToDrop).take(page.size).list
+    (for (r <- rows if r.ownerId === userId && r.state === LibraryStates.ACTIVE) yield r).sortBy(x => (x.kind, x.memberCount.desc, x.lastKept.desc, x.id.desc)).drop(page.itemsToDrop).take(page.size).list
```
Comment (2is10) [inline kifi-backend/modules/shoebox/app/com/keepit/model/LibraryRepo.scala:179]: How fortunate we are that `.kind` works the way we want here! :smile:

File: kifi-backend/modules/shoebox/test/com/keepit/commanders/LibraryCommanderTest.scala [hunk 2]
```diff
@@ -48,6 +48,18 @@ class LibraryCommanderTest extends TestKitSupport with SpecificationLike with Sh
...
+      (self.kind.value compare that.kind.value) match {
+        case 0 =>
+          (self.memberCount compare that.memberCount) match {
+            case 0 => -(self.id.get.id compare that.id.get.id)
+            case c => -c
+          }
+        case c => c
+      }
+  }
+
```
Comment (2is10) [inline kifi-backend/modules/shoebox/test/com/keepit/commanders/LibraryCommanderTest.scala:62]: very nice!

**Comment by 2is10** [comment]
:+1:

File: kifi-backend/modules/shoebox/app/com/keepit/model/LibraryRepo.scala (hunk 1 above)
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/app/com/keepit/model/LibraryRepo.scala:179]: haha yea! :dog:

**Pull request merged**